src/hooks/useChartPagination.js

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react';
+import { getCharts } from '@apis/chartsApi';
+
+/**
+ * 차트 리스트 페이지네이션을 위한 커스텀 훅
+ *
+ * @param {'male' | 'female'} gender - 조회할 차트 성별 (남자/여자)
+ * @returns {{
+ *   chartDataList: Array,              // 현재까지 불러온 차트 데이터 리스트
```

Comment:
구조를 다음과 같이 조금 더 구체화하면 VSCode, JSDoc 인텔리센스 활용 시 툴링 측면에서 좋습니다.
```suggestion
 *   chartDataList: Array<Chart>,              // 현재까지 불러온 차트 데이터 리스트
```